backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java

```diff
@@ -314,66 +360,40 @@ public RepositoryEntity getRepositoryById(Session session, RepositoryIdentificat
             "repositoryName",
             repository.getName(),
             ModelDBConstants.WORKSPACE_ID,
-            workspaceDTO.getWorkspaceId(),
-            workspaceDTO.getWorkspaceType(),
+            repositoryEntity.getWorkspace_id(),
+            WorkspaceType.forNumber(repositoryEntity.getWorkspace_type()),
             LOGGER);
-        repositoryEntity = new RepositoryEntity(repository, workspaceDTO);
-        repositoryEntity.setDeleted(true);
-      } else {
-        repositoryEntity = getRepositoryById(session, request.getId(), true);
-        if (!repository.getName().isEmpty()
-            && !repositoryEntity.getName().equals(repository.getName())) {
-          ModelDBHibernateUtil.checkIfEntityAlreadyExists(
-              session,
-              SHORT_NAME,
-              GET_REPOSITORY_COUNT_BY_NAME_PREFIX_HQL,
-              RepositoryEntity.class.getSimpleName(),
-              "repositoryName",
-              repository.getName(),
-              ModelDBConstants.WORKSPACE_ID,
-              repositoryEntity.getWorkspace_id(),
-              WorkspaceType.forNumber(repositoryEntity.getWorkspace_type()),
-              LOGGER);
-        }
-        repositoryEntity.update(request);
       }
+      repositoryEntity.update(request);
+    }
+    session.beginTransaction();
+    session.saveOrUpdate(repositoryEntity);
+    if (create) {
+      Commit initCommit =
+          Commit.newBuilder().setMessage(ModelDBConstants.INITIAL_COMMIT_MESSAGE).build();
+      CommitEntity commitEntity =
+          commitDAO.saveCommitEntity(
+              session,
+              initCommit,
+              FileHasher.getSha(new String()),
+              authService.getVertaIdFromUserInfo(userInfo),
+              repositoryEntity);
+
+      saveBranch(
+          session, commitEntity.getCommit_hash(), ModelDBConstants.MASTER_BRANCH, repositoryEntity);
+    }
+    session.getTransaction().commit();
+    if (create) {
+      createRoleBindingsForRepository(request, userInfo, repositoryEntity);
+
+      // Update repository deleted status to false after roleBindings created successfully
       session.beginTransaction();
-      session.saveOrUpdate(repositoryEntity);
-      if (create) {
-        Commit initCommit =
-            Commit.newBuilder().setMessage(ModelDBConstants.INITIAL_COMMIT_MESSAGE).build();
-        CommitEntity commitEntity =
-            commitDAO.saveCommitEntity(
-                session,
-                initCommit,
-                FileHasher.getSha(new String()),
-                authService.getVertaIdFromUserInfo(userInfo),
-                repositoryEntity);
-
-        saveBranch(
-            session,
-            commitEntity.getCommit_hash(),
-            ModelDBConstants.MASTER_BRANCH,
-            repositoryEntity);
-      }
+      repositoryEntity.setDeleted(false);
```

Comment:
please pull in master to update on the repo creation logic

Reply:
pulled master